ttnn/cpp/ttnn/operations/ccl/kernel_common/worker_sync_utils.hpp

```diff
@@ -230,22 +227,15 @@ struct MatmulOpReceiver {
                 reinterpret_cast<volatile tt_l1_ptr uint32_t*>(get_semaphore(get_arg_val<uint32_t>(rt_args_idx++)));
         }
 
-        this->num_tensor_slices = this->num_transfers * this->num_directions;
-
-        // Setup internal states for bi-direction
-        this->ring_idxs[0] = start_ring_index;
-        this->ring_idxs[1] = start_ring_index;
-
-        this->start_page_idxs[0] = this->ring_idxs[0] * this->output_page_offset;
-        this->start_page_idxs[1] = this->ring_idxs[1] * this->output_page_offset;
-
-        this->is_clockwise_dirs[0] = is_clockwise_direction;
-        this->is_clockwise_dirs[1] = !is_clockwise_direction;
+        // Setup internal states
+        for (uint32_t i = 0; i < this->num_directions; i++) {
+            this->ring_idxs[i] = start_ring_index;
+            this->start_page_idxs[i] = this->ring_idxs[0] * this->output_page_offset;
```

Comment:
If `ring_idxs[i]` is always `start_ring_index` and `start_page_idxs[i] is always `this->ring_idxs[0] * this->output_page_offset` i.e. `start_ring_index * this->output_page_offset` for all values of `i`, why bother with the arrays at all?

Why not just have single variables?